includes/class-item-base.php

```diff
@@ -66,12 +66,12 @@ function __construct( $args = array() ) {
 
 				}
 
-				if ( method_exists( $class_name, 'make_new' ) ) {
+				if ( method_exists( $class_name, 'make_new_item' ) ) {
 					/*
 					 * If this class has a make_new() method.
-					 * A Class::make_new() method is needed when the class constructor has more than 1 parameter of $args.
```

Comment:
@sarutole Looking at it more, `make_new_item()` is used when the class that is making the item is **_not_** an `WPLib_Item_Base` but wants to create an instance of one, e.g. `WPLib_Posts` or a subclass of `WPLib_Module_Base`.

`make_new()` is instead used to create instances of the _**current**_ class, e.g. instances of `WPLib_Item_Base`, `WPLib_Model_Base`, `WPLib_View_Base`, `WPLib_User` etc.

I fear what has happened is that was not well understood _(given lack of documentation)_ and our prior code may have misused those names?  

BTW, if I accepted the above pull request it would create a situation where a Model or View could not follow conventions to have a method can creates an instance of `WPLib_Item_Base`.  Can you share with me your use-case code that triggers this pull request?